### ISSUE TITLE
Telegram desktop asks about access code every time I start Telegram application

### ISSUE THREAD

**Issue by mareg** [telegramdesktop/tdesktop#960]
OS: Fedora 22 64bit

Every day I come from work and I start telegram on my desktop - it asks to give my phone number and requests the access code. It forgets notification settings as well. As if it didn't save the data on disk.

On my work laptop it remembers settings between sessions. No idea why it happens, I run Fedora 22 64bit on my work laptop too.

**Comment by fjfalcon** [comment]
Got same issue, archlinux 64, telegram-desktop installed from aur. 
Strange fact, issue starts when i get tdesktop updated.

**Comment by telegramdesktop** [comment]
installed from aur? what if you try binary from tdesktop.com?

looks like it can't write on disk, yes.. it tries to write to ~/.TelegramDesktop, could it be write protected from him?

**Comment by fjfalcon** [comment]
So, after i installed binary from tdesktop.com telegram asked about access code only once. And all works as it should. 
Before that i checked acl with
`find ./TelegramDesktop -type f -exec ls -l {} \;` 
and all files and have my local user as owner. 
That quite strange...

**Comment by telegramdesktop** [comment]
I'm not sure then, what build have you used before, I know only about the binary build from tdesktop.com, which I test in several Linux distros, that it should work and save all its files in home dir in ".TelegramDesktop" hidden folder.

Does topic starter use binary from official site or some other installation? @mareg

**Comment by mareg** [comment]
For me - sometimes it survives the restart, other times it doesn't. but it happens only on one computer, on the other computer it works just fine. there's no pattern.

@telegramdesktop I am using `rommon-telegram` repository hosted on fedoradesktop.org - repositories are standard ways of managing updates on linux platform, I do not install builds downloaded off the web, I only install applications if they provide repo for their packages.

I can tell that Telegram uses some kind of auto-update mechanism - this is not how it should be done on linux, because I don't have control which version I am running. Yes I do install updates (my OS checks for them daily).

And yes, `.TelegramDesktop` is writable:
```
$ ll -a | grep Telegram
drwxrwxr-x    4 marek marek       78 Aug  6 10:58 .TelegramDesktop
```

**Comment by alleyoopster** [comment]
Mine is installed in /opt and gets updates through the app. Same problem started a few days ago. I'm on Mint 17.2 (Ubuntu 14.04 base). The directory is writeable for the user.

**Comment by mareg** [comment]
@alleyoopster The new update is being downloaded and installed into `~/.TelegramDesktop` which may be OK on Windows platform, but is desired on Linux platform.

**Comment by mareg** [comment]
@telegramdesktop having to enter the code every time I start Telegram is really annoying.

**Comment by alleyoopster** [comment]
@mareg It is working for me again now. maybe since the last update. I don't think it matters where it is installed as long as the directory is writeable for the app.

**Comment by mareg** [comment]
@alleyoopster it's on and off for me - usually when the update arrives, it "forgets" settings.
Also I have two entries in the menu now "Telegram" and "Telegram Desktop"... one pointing to the version installed through package manager, other pointing to the latest version installed via updater in my user's directory. Hope this PR https://github.com/telegramdesktop/tdesktop/pull/920 will solve the auto updater issue.

**Comment by auchri** [comment]
#1020 was just merged :)

**Comment by telegramdesktop** [comment]
Well, new builds for repositories now can disable auto-update from tdesktop.com by a macro, I'm afraid I won't fix anything else for them, cause I test only builds that are available at https://desktop.telegram.org for download.

**Comment by fjfalcon** [comment]
Ok, i ask you to reopen this bug. 

Telegram version from desktop.telegram.org. 
Bugs: 
Session dropped, i almost always need to enter code from my mobile phone.
This especialy encounters when i recieve new update, cause i recieve it multiple times for same version. Dunno why. 
Telegram directory fully writable, i especialy do 777 for testing purprose. 

OS: ArchLinux 
Telegram version: 0.9.2 

My chat with service notifications... It's really annoying. 
If there any more options i can provide to help you fix this annoying bug - please, tell me.
![2015-09-27_566x1062](https://cloud.githubusercontent.com/assets/1033431/10123430/ffca9b5c-6541-11e5-80a7-5e0b41f6807c.png)

**Comment by alleyoopster** [comment]
@fjfalcon I feel your frustration. I had this for a while also. Have you tried uninstalling completely (check home hidden folders and /opt) and installing telegram bin from AUR. After this let Arch do the updates from AUR and turn off auto updates in the Telegram app. I am running like this and all is good.

**Comment by Quarx2k** [comment]
I had same on windows with 0.8.x 0.9.1, when telegram freezes and i kill process with Task Manager. After next start, app ask for phone number again and all settings cleared.
On Ubuntu 15.04 too, sometimes, when i forgot correct exit from app before shutdown pc.

**Comment by andlabs** [comment]
This is happening for me on Ubuntu 15.04 with 0.9.2 as well.

**Comment by telegramdesktop** [comment]
@andlabs Telegram Desktop has all its data in ~/.TelegramDesktop folder, it should be available for write. What is in log.txt from that folder?

**Comment by andlabs** [comment]
```
[2015.09.30 13:07:55] Audio init time: 48
[2015.09.30 13:07:56] TCP Error: network - Network unreachable
[2015.09.30 13:07:56] HTTP Error: network error 99 - Network unreachable
[2015.09.30 13:07:56] Not using Unity Launcher count.
[2015.09.30 13:07:57] TCP Error: network - Network unreachable
[2015.09.30 13:07:57] HTTP Error: network error 99 - Network unreachable
[2015.09.30 13:08:00] TCP Error: network - Network unreachable
[2015.09.30 13:08:00] HTTP Error: network error 99 - Network unreachable
[2015.09.30 13:08:01] TCP Error: network - Network unreachable
[2015.09.30 13:08:01] HTTP Error: network error 99 - Network unreachable
[2015.09.30 13:08:04] TCP Error: network - Network unreachable
[2015.09.30 13:08:04] HTTP Error: network error 99 - Network unreachable
[2015.09.30 13:08:05] TCP Error: network - Network unreachable
[2015.09.30 13:08:05] HTTP Error: network error 99 - Network unreachable
```

**Comment by andlabs** [comment]
I figured it out: I still had the 0.8 binary in my $PATH, and it was loading that and getting confused. I changed the $PATH entry to be a symlink to ~/.TelegramDesktop/Telegram and now everything works.

**Comment by telegramdesktop** [comment]
@andlabs There is no need to gold Telegram binary in ~/.TelegramDesktop, this folder is used for storing data and should be accessible for writing by Telegram binary, but the binary itself can be in any other place, like ~/Desktop/Telegram/ for example. In order for autoupdater to work it should be accessible for writing as well.

**Comment by andlabs** [comment]
All right, good to know. I'm not sure why the updater didn't overwrite my ~/bin/Telegram, so I'll have to look into this more sometime...

**Comment by RomanHargrave** [comment]
@telegramdesktop Nope, the updater did not appear to "update" the telegram binary at `~/.local/bin/Telegram`.

**Comment by TheBlackViper1996** [comment]
I'm having the same issue, this is what the log file says:
[2015.12.14 02:01:35] RPC Error: request 14 got fail with code 400, error AUTH_BYTES_INVALID
[2015.12.14 02:01:36] RPC Error: request 16 got fail with code 401, error AUTH_KEY_UNREGISTERED
[2015.12.14 02:01:36] RPC Error: request 18 got fail with code 401, error AUTH_KEY_UNREGISTERED
[2015.12.14 02:01:37] RPC Error: request 17 got fail with code 401, error AUTH_KEY_UNREGISTERED
[2015.12.14 02:01:37] RPC Error: request 19 got fail with code 401, error AUTH_KEY_UNREGISTERED
[2015.12.14 02:03:16] RPC Error: request 66 got fail with code 400, error AUTH_BYTES_INVALID
[2015.12.14 02:03:16] RPC Error: request 69 got fail with code 401, error AUTH_KEY_UNREGISTERED
[2015.12.14 02:03:17] RPC Error: request 70 got fail with code 401, error AUTH_KEY_UNREGISTERED
[2015.12.14 02:03:17] RPC Error: request 68 got fail with code 401, error AUTH_KEY_UNREGISTERED
[2015.12.14 02:03:17] RPC Error: request 71 got fail with code 401, error AUTH_KEY_UNREGISTERED

**Comment by fjfalcon** [comment]
Problem still not solved, btw. So, it's nice you reopen it.

**Comment by serviceman** [comment]
solved for me with the "hack" of @andlabs

**Comment by lucaciavatta** [comment]
Got same issue on Fedora 23 with Telegram Desktop 0.9.33 and log file shows this:
`[2016.03.20 06:38:48] Could not load 'appindicator3' with version 1 :(
[2016.03.20 06:38:48] Could not load 'appindicator' with version 1 :(
[2016.03.20 06:38:48] App Info: reading settings..
[2016.03.20 06:38:48] App Info: reading encrypted settings..
[2016.03.20 06:38:48] App Info: reading map..
[2016.03.20 06:38:48] App Info: reading encrypted map..
[2016.03.20 06:38:48] App Info: reading encrypted user settings..
[2016.03.20 06:38:48] App Info: reading encrypted mtp data..
[2016.03.20 06:38:48] Map read time: 0
[2016.03.20 06:38:48] Not using Unity Launcher count.
[2016.03.20 06:38:48] TCP Error: network - Network unreachable
[2016.03.20 06:38:48] HTTP Error: network error 99 - Network unreachable
[2016.03.20 06:39:14] TCP Error: network - Network unreachable
[2016.03.20 06:39:14] TCP Error: network - Network unreachable
[2016.03.20 06:39:14] HTTP Error: network error 99 - Network unreachable
[2016.03.20 06:43:59] Message Info: bad message notification received (error_code 16) for msg_id = 6264005057608610020, seq_no = 57
[2016.03.20 06:44:14] TCP Error: network - Network unreachable
[2016.03.20 06:44:14] HTTP Error: network error 99 - Network unreachable`

**Comment by algui91** [comment]
I still have the same problem. :-(

**Comment by fjfalcon** [comment]
As for me i have this workaround: 
I create link to ~/Telegram/telegram to /usr/bin/telegram so i can easy run in. 
When telegram ask for update i simple close him, going to telegram.org, downloading it, simple tar xvf tsetup...tar.xz and my ~/Telegram folder gets updated.
As for you you can create .desktop file or any other pointer to run telegram from your favorite launcher.
No more problem with update.

**Comment by algui91** [comment]
I have the bin in ~/bin/Telegram/ and in that folder I have the binary and
the updater,
before, I do not know why, in .TelegramDesktop/ I also have the updater and
the binary, I just removed .TelegramDesktop and reinstall Telegram, hope
this solve the problem.

On Sun, Mar 20, 2016 at 8:57 AM Pavel Chernyak <notifications@github.com>
wrote:

> As for me i have this workaround:
> I create link to ~/Telegram/telegram to /usr/bin/telegram so i can easy
> run in.
> When telegram ask for update i simple close him, going to telegram.org,
> downloading it, simple tar xvf tsetup...tar.xz and my ~/Telegram folder
> gets updated.
> As for you you can create .desktop file or any other pointer to run
> telegram from your favorite launcher.
> No more problem with update.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/telegramdesktop/tdesktop/issues/960#issuecomment-198868769>
>
-- 

Alejandro Alcalde - elbauldelprogramador.com

**Comment by Zelmor** [comment]
I am experiencing this issue as well. Any updates on this one?

**Comment by fjfalcon** [comment]
Well, i simple don't use telegram autoupdate and update it manualy when client finds new version.

**Comment by algui91** [comment]
I solve it moving the telegram executable inside que folder. .Telegram,
and then adding that directory to my path, no problems since then.

On Sat, Jun 11, 2016, 23:39 Pavel Chernyak <notifications@github.com> wrote:

> Well, i simple don't use telegram autoupdate and update it manualy when
> client finds new version.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/telegramdesktop/tdesktop/issues/960#issuecomment-225395974>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAYGLZcw-3NO3G8tjp88LLOVHbC8mdR1ks5qKysbgaJpZM4FljkQ>
> .
>
-- 

Alejandro Alcalde - elbauldelprogramador.com

**Comment by sindrom91** [comment]
I can confirm that Telegram didn't update the actual binary I ran, but it did add the updated binary to `~/.TelegramDesktop`.

**Comment by sindrom91** [comment]
I created a symbolic link to Telegram binary in `/usr/local/bin/` and that is what I use to run Telegram. When I run telegram like that, it doesn't update properly. But when I use the actual binary to run it, it updates properly. I also noticed that, when I ran it from symbolic link, `Executable dir` in `log.txt` was set to `~/.TelegramDesktop` (that is not where the binary is located).

**Comment by SharkWipf** [comment]
Ran into this as well, until an official fix is created for this, the easiest workaround would probably be to create a symlink to the binary that does properly get updated, so doing something like: 
`ln -fs ~/.TelegramDesktop/Telegram ~/bin/telegram`
(replace ~/bin/telegram with wherever you have your symlink, i.e. /usr/local/bin/telegram). 

No moving of executables or whatever required otherwise, just a new symlink.

**Comment by sindrom91** [comment]
I found another workaround that works for me. I created a script that runs Telegram, put it in `/usr/local/bin` and gave it execution permission. It seems to work well, but there has been just one update, and I didn't have any issues.

The script:
```
#!/bin/bash
cd /path/to/Telegram && ./Telegram
```

**Comment by Zelmor** [comment]
For me the issue arises after update. My Telegram application is located in /opt/Telegram and a symlink is created for /bin/telegram

/opt is where this application should, in my opinion, store its binaries. Not under the home folder.

**Comment by strasharo** [comment]
Same here on Fedora 24. There are two shortcuts for the app.
One is global and is located in /usr/share/applications/telegram-desktop.desktop
[Desktop Entry]
Name=Telegram
GenericName=Telegram Messenger
**Exec=telegram**
Terminal=false
Icon=telegram
Type=Application
Categories=Network;
X-Desktop-File-Install-Version=0.22

After the update the following shortcut gets created:
/home/<user>/.local/share/applications/telegramdesktop.desktop

[Desktop Entry]
Encoding=UTF-8
Version=1.0
Name=Telegram Desktop
Comment=Official desktop version of Telegram messaging app
**Exec=/home/<user>/.TelegramDesktop/Telegram -- %u**
Icon=telegram
Terminal=false
StartupWMClass=Telegram
Type=Application
Categories=Network;
MimeType=x-scheme-handler/tg;
X-Desktop-File-Install-Version=0.23

And it points to the updated version located in the user's home directory.
So we end up with two shortcuts in the menu - one wrong and one correct.

This causes the mismatch and the described behavior.

**Comment by 5chdn** [comment]
Hell, this is an annoying issue. After reading this whole thread finally, I can only suggest to _disable_ auto update on linux systems _by default_. In opposite to systems like osx or windows, you usually get updates of the software from the repositories. Please consider this, it can't be expected of all linux users to disable it manually or hack the $PATH.

**Comment by SharkWipf** [comment]
I'd say a better fix would be to just make the executed binary execute the right, updated binary immediately, that way the (possibly read-only) executed binary can still be executed and the (r/w) binary in the user's home folder can still be updated.
In fact, the only thing the downloaded binary would have to be is an updater/launcher, I don't see any reason why it should be doing anything more than that, the new binary is put into the .TelegramDesktop folder anyway.

The traditional way of relying on a package manager for updates is a bit of an issue.
Most non-rolling distros stick to one major version of a program for the lifetime of their release, which is fine for server software, but for rapidly-updating internet facing desktop software it is a bit troublesome. 
Telegram would have to remain backwards compatible until distros are EOL, or force users of those distros to use a self-installed version, defeating the point of the package manager.

I'd suggest making the main application purely be an updater/launcher, which would fix this problem, and make packaging it easier as well.

**Comment by john-preston** [comment]
@5chdn Why don't you just download+unpack the version to some ~/tdesktop folder and launch it from there? The updater works fine that way.

**Comment by 5chdn** [comment]
>  Most non-rolling distros stick to one major version of a program for the lifetime of their release, which is fine for server software, but for rapidly-updating internet facing desktop software it is a bit troublesome. 

Never understood non-rolling releases unless you run a server.

> I'd suggest making the main application purely be an updater/launcher, which would fix this problem, and make packaging it easier as well.

Can not see how this works out regarding permissions. Telegram has no permissions to upgrade software on my computer, and will never have for obvious reasons.

> Why don't you just download+unpack the version to some ~/tdesktop folder and launch it from there?

For the lack of convenience, I can not keep track of all packages installed on my systems and therefore I'm using a package manager like every linux distribution offers. Installing telegram like any other package from the repositories allows me just to use it without messing my system with custom path and binary locations.

After keeping logged out of telegram on all my machines each day, I came here to suggest two simple solutions which are both (in my eyes) trivial and not messy:

- disable auto update in linux environments by default
- keep logins valid through different versions

If you do not want to do this, I'm totally fine with manually disabling auto updates on all my devices. However, I see a lot of pain in this thread and just wanted to be constructive.

**Comment by SharkWipf** [comment]
> Never understood non-rolling releases unless you run a server.

A lot less tech-savvy Linux desktop users will be stuck to distros like Ubuntu, Mint or Fedora. Arch, Gentoo, Debian Sid, etc are not an option for them. At most LMDE.
> Can not see how this works out regarding permissions.

Exactly the same as it works now, the launched binary just updates the `~/.TelegramDesktop/Telegram` file like it already does, but then launches that and exits, instead of current behaviour.

> - keep logins valid through different versions

That might be less trivial than it originally sounds, and is quite possibly the original reason they opted for auto-updating over depending on package managers in the first place. Protocols may (have to) change, regardless of reason. Sometimes backwards compatibility is not an option due to security issues, or pushes extra, unnecessary workload on the developers.

IMO, in some cases, disabling automatic updates, or at least disabling built-in updaters and relying only on package managers, can be a bigger security risk than just letting the application update itself (with care, of course).

**Comment by john-preston** [comment]
@5chdn I'm not sure what are the binaries provided by your repositories and where do they come from, but if it is a custom build of telegram desktop from the source files located here the author of the package can easily disable auto update in the source code by a TDESKTOP_DISABLE_AUTOUPDATE macro.

**Comment by rhoit** [comment]
i'm using `telegram-desktop-bin-dev` having same problem.

**Comment by cookiengineer** [comment]
Had same problem using the `telegram-desktop-bin` and `telegram-desktop-bin-dev` from AUR. Switched to the official binaries from the website and just moved it to `/opt/telegram/Telegram` and symlinked it.

Worked fine with binaries, but I guess I will get updating problems in future.

**Comment by Suseika** [comment]
Had the same issue, https://github.com/telegramdesktop/tdesktop/issues/960#issuecomment-146177958 solved it for me. I had my `Telegram` binary at `~/.bin/Telegram`. I changed it to a symlink to ~/.TelegramDesktop/Telegram`, and now it works properly.

**Comment by atomAltera** [comment]
Any updates here?

**Comment by john-preston** [comment]
@atomAltera 1. What OS do you use?

2. Where is Telegram binary located?

3. Do you use Telegram from https://desktop.telegram.org or from some other installation?

**Comment by atomAltera** [comment]
@john-preston I'm using BunsenLabs (based on Debian 8). 
Yes, I downloaded official binaries from https://desktop.telegram.org/  (Telegram and Updater)
They are located in ~/opt/Telegram/ and I have symlink from ~/bin/telegram -> ~/opt/Telegram/Telegram

**Comment by rugubara** [comment]
versions 1.1.9 and 1.1.10 exhibit the same problem for me.  I've removed ~/.TelegramDesktop and restarted the client. the directory was not re-created. I'm being asked for the verification code every time I start the app. I'm on Gentoo and telegram is part of the main repository. I've checked the source url of the ebuild - it's from updates.tdesktop.com

**Comment by john-preston** [comment]
@rugubara Please try to download version from https://desktop.telegram.org and use it from somewhere inside your home folder, it should work.

**Comment by rugubara** [comment]
correction - 1.1.9 is affected. 1.1.10 is not. Thanks.

**Comment by john-preston** [comment]
@rugubara I'm afraid the problem is not in the app version, but in autoupdate. When 1.1.10 gets updated to the next version it will logout as well.

But the version from site placed in some place with write access should not.

**Comment by theSimplex** [comment]
Still having this issue, tried a bunch of stuff from the thread - won't fix the issue.
(Fedora26)

**Comment by lucysan** [comment]
Still having same issue, using telegram-desktop from aur, running Arch

**Comment by john-preston** [comment]
@lucysan This should not happen if you take the version from https://desktop.telegram.org and unpack it to some write-access-enabled folder in your home directory, like on Desktop.

**Comment by kaldown** [comment]
@john-preston it's not true.
```
ls -l ~/.local/Telegram 
total 74708
-rwxr-xr-x 1 kaldown kaldown 76297256 Jul  4 19:50 Telegram
-rwxr-xr-x 1 kaldown kaldown   196723 Jul  4 19:24 Updater
```
Every time ask fro an update.
Every time have prev version after reboot.
Every time ask a code.
On two machines remains same flow.

**Comment by john-preston** [comment]
@kaldown I can't reproduce that :( can you enable debug mode in Settings by typing "debugmode" and try to auto update to collect logs after that?

**Comment by kaldown** [comment]
@john-preston there is no "debugmode" in settings of the app.

**Comment by john-preston** [comment]
@kaldown Type "debugmode" from your keyboard, it should offer you to switch the debug logs on.

**Comment by kaldown** [comment]
so where thouse logs should appers

**Comment by john-preston** [comment]
@kaldown In ~/.TelegramDesktop/DebugLogs or ~/.local/share/TelegramDesktop/DebugLogs

**Comment by kaldown** [comment]
As I read in logs, Telegram somehow think that my workdir in `~/.local/share/TelegramDesktop`
but app were laucned from `~/.local/bin/Telegram/Telegram`, which is in my $PATH.
Now I see that `~/.local/share/TelegramDesktop/Telegram` is more recent version.
But I won't my app be there. 
How to change workdir?

**Comment by john-preston** [comment]
@kaldown workdir doesn't change and it should be:

 ~/.local/share/TelegramDesktop

but the executable should not be there, it should be in the place you placed it and it should autoupdate there. I ask for update logs (text files in DebugLogs containing "upd") to see how this happens.

**Comment by kaldown** [comment]
https://pastebin.com/Dvj6ELGr

**Comment by john-preston** [comment]
@kaldown Thanks! I see, it somehow gets an empty string in "Updater binary dir". I'll try to investigate that.

**Comment by john-preston** [comment]
@kaldown Not the full launch path is given in the argument for Updater =( I'll attempt to fix that in the next version.

**Comment by NuLL3rr0r** [comment]
I have the exact same issue with Telegram desktop on Funtoo Linux; on both my work and home PC's. It used to work as expected. It has been some time since this issue has appeared. Very frustrating. I have to re-activate on each start. It does not matter if I download from telegram.org or emerge it from Portage.

I append debugmode to the Telegram command line but cannot get any debug logs inside ~/.local/share/TelegramDesktop either.

**Comment by john-preston** [comment]
@NuLL3rr0r @kaldown 

And how do you launch the app? It looks like I don't get the executable path in the first argument (argv[0]). Looks like when I don't get the full path to the executable bad things happen :(

**Comment by john-preston** [comment]
@NuLL3rr0r Can you also send corresponding "log.txt" for such launch? So I can see the real arguments being passed.

**Comment by NuLL3rr0r** [comment]
@john-preston I uninstalled Telegram binary from portage then purged everything inside ~/.local/share/TelegramDesktop. Then re-activated using the downloaded binary from telegram.org. Hopefully, the issue is gone.

Sorry, I don't have the log file anymore. But, it looks like if you have (at least on Funtoo Linux) Telegram installed from your package manager it won't work (since it resides inside /usr/bin) and afterwards if you run the downloaded binary from telegram.org it happens to both binaries; unless you start from scratch as I did.

**Comment by john-preston** [comment]
@NuLL3rr0r I will fix that in our binary. I hope package managers, whoever they are, will stop deploying wrong-working builds :(

**Comment by fjfalcon** [comment]
Wow, 2 years bug fixed.
Thank you @john-preston

**Comment by john-preston** [comment]
This still needs testing, I could not reproduce that, so I tried to fix it blindly :/

**Comment by bdod** [comment]
[2017.11.10 11:18:25] Audio Playback Devices: OpenAL Soft
[2017.11.10 11:18:25] Audio Playback Default Device: OpenAL Soft
[2017.11.10 11:18:25] Audio Capture Devices: Встроенное аудио Аналоговый стерео;Monitor of Встроенное аудио Аналоговый стерео
[2017.11.10 11:18:25] Audio Capture Default Device: Встроенное аудио Аналоговый стерео
[2017.11.10 11:18:25] TCP Error: network - Network unreachable
[2017.11.10 11:18:25] HTTP Error: network error 99 - Network unreachable
[2017.11.10 11:18:25] Tray Icon: Using appindicator tray icon.
[2017.11.10 11:18:25] Found Unity Launcher entry telegramdesktop.desktop!
[2017.11.10 11:18:25] TCP Error: network - Network unreachable
[2017.11.10 11:18:25] HTTP Error: network error 99 - Network unreachable
[2017.11.10 11:18:36] Lang Error: Unknown key 'lng_channel_admin_sure'
[2017.11.10 11:18:36] Lang Error: Unknown key 'lng_context_remove_admin'

таже фигня (( ...

**Comment by linuxtopia** [comment]
I wonder if shutdown pc(win10) then Telegram would ask phone number?

**Comment by h3ct0rjs** [comment]
Hi, after reading all the comments above it seems that it persist, is there anyway to get a full log o debug details. I've my telegram folder as @gvlasov  suggest.

**Comment by h3ct0rjs** [comment]
After reinstalling again in the  folders previously mentioned by @gvlasov it seems that it was solved. However I would like to learn how to get a log or debug log of telegram. 

Best Regards, 
Hector F.

**Comment by codywohlers** [comment]
Everytime I close telegram I have to log back in.  Workaround was to make a symlink for 
`~/.TelegramDesktop -> ~/.local/share/TelegramDesktop`

KUbuntu 20.04.1 LTS focal
Linux 5.4.0-58-generic x86_64
plasmashell 5.18.5
kwin 5.18.5
Qt: 5.12.8
KDE Framework: 5.68.0
